src/main/java/Command.java

```diff
@@ -0,0 +1,18 @@
+public abstract class Command {
+    protected String command;
+    protected int index;
+    protected Task task;
```

Comment:
Maybe you can consider having this 'final' as you won't and probably shouldn't be able to change them.